### PR TITLE
Jonmv/misc 6

### PR DESCRIPTION
**Pull request by jonmv**
@mpolden please review.

### CODE REVIEW

**Comment by jonmv** [comment]
I think we should cover existing application names with this, but I'm not entirely sure. 
If we think it's too sketchy, I can go through the code and fix the application-name nonsense first.

**Comment by jonmv** [comment]
Is this too risky, @bratseth?

**Comment by jonmv** [comment]
I can't fix the application name thingy. It's optionally used when preparing sessions. We use ruby tests name as application name in our system tests. I suppose others could use anything, but I think it's not very likely? We could just disallow the things that break serialisation, though, like `:` and `/` ...

File: config-provisioning/src/main/java/com/yahoo/config/provision/ApplicationId.java [hunk 1]
```diff
@@ -12,24 +12,26 @@
...
  */
-public final class ApplicationId implements Comparable<ApplicationId> {
+public class ApplicationId implements Comparable<ApplicationId> {
 
     private final TenantName tenant;
     private final ApplicationName application;
     private final InstanceName instance;
     private final String serializedForm;
 
-    public ApplicationId(ApplicationIdConfig config) {
```
Comment (bratseth) [inline config-provisioning/src/main/java/com/yahoo/config/provision/ApplicationId.java:22]: This API is used by old config models, so you need to be sure they are not using this. I think it is ok, but not sure it's worth the risk just to make final implicit?

**Review by bratseth — APPROVED** [review]
Great, I really think we should do this and that we don't have any actual applications breaking these rules.

File: config-provisioning/src/main/java/com/yahoo/config/provision/ApplicationId.java (hunk 1 above)
Comment (jonmv) [reply config-provisioning/src/main/java/com/yahoo/config/provision/ApplicationId.java:22]: I could only find two usages of this: a unit test, and the `NginxMetricsReporter` component.

**Pull request merged**